Index: postgres/create_funcs.sql
===================================================================
--- postgres/create_funcs.sql	(revisjon 2238)
+++ postgres/create_funcs.sql	(arbeidskopi)
@@ -120,6 +120,25 @@
 END;
 $$ LANGUAGE plpgsql;
 
+DROP FUNCTION hvor(timestamptz);
+CREATE OR REPLACE FUNCTION hvor(currtime timestamptz) RETURNS SETOF timestamptz AS $$
+DECLARE
+    currpos point;
+    currsted text;
+    currdist numeric;
+    currlat numeric(9, 6);
+    currlon numeric(9, 6);
+BEGIN
+    currpos = findpos(currtime);
+    currlat = currpos[0];
+    currlon = currpos[1];
+    currsted = clname(currpos);
+    currdist = cldist(currpos);
+    SELECT currtime, currlat::text, currlon::text, currsted, currdist;
+    RETURN;
+END;
+$$ LANGUAGE plpgsql;
+
 DROP FUNCTION wherepos(timestamptz);
 CREATE OR REPLACE FUNCTION wherepos(currtime timestamptz) RETURNS text AS $$
 DECLARE
