diff --git a/.gitmodules b/.gitmodules
index a293c5087c09..0b4ad0978ddd 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,18 +1,18 @@
 [submodule "Git/gpstools"]
 	path = Git/gpstools
-	url = https://gitlab.com/sunny256/gpstools.git
+	url = sunny@bellmann:~/bin/Git/gpstools
 	branch = master
 [submodule "Git/suuid"]
 	path = Git/suuid
-	url = https://gitlab.com/sunny256/suuid.git
+	url = sunny@bellmann:~/bin/Git/suuid
 	branch = master
 [submodule "Git/svnutils"]
 	path = Git/svnutils
-	url = https://gitlab.com/sunny256/svnutils.git
+	url = sunny@bellmann:~/bin/Git/svnutils
 	branch = master
 [submodule "Git/spar"]
 	path = Git/spar
-	url = https://gitlab.com/sunny256/spar.git
+	url = sunny@bellmann:~/bin/Git/spar
 [submodule "Git/dropboxannex"]
 	path = Git/dropboxannex
-	url = https://gitlab.com/sunny256/dropboxannex.git
+	url = sunny@bellmann:~/bin/Git/dropboxannex
