Index: tests/run-tests.pl
===================================================================
--- tests/run-tests.pl	(revision 1843)
+++ tests/run-tests.pl	(working copy)
@@ -828,15 +828,17 @@
 <track>
 <title>Missing various elements</title>
 <tp> <time>2006-04-30T17:16:59Z</time> </tp>
-<tp> <time>2006-04-30T17:17:00Z</time> </tp>
+<tp> <time>2006-04-30T17:17:00Z</time> <lat>60.42352</lat> </tp>
 <tp> <time>2006-04-30T17:17:09Z</time> <lat>60.42353</lat> <lon>5.34185</lon> </tp>
-<tp> <time>2006-04-30T17:17:11Z</time> <ele>483</ele> </tp>
+<tp> <time>2006-04-30T17:17:11Z</time> <lon>5.34187</lon> <ele>483</ele> </tp>
 <tp> <time>2006-04-30T17:17:22Z</time> <ele>485</ele> </tp>
 <tp> <lat>60.42347</lat> <lon>5.34212</lon> <ele>486</ele> </tp>
-<tp> <ele>484</ele> </tp>
+<tp> <lon>5.34224</lon> <ele>484</ele> </tp>
 <tp> <ele>486</ele> </tp>
 <tp> <desc>Missing everything</desc> </tp>
-<tp> <time>2006-04-30T17:18:03Z</time> <ele>490</ele> </tp>
+<tp> <lat>60.42339</lat> </tp>
+<tp> <lon>5.34262</lon> </tp>
+<tp> <time>2006-04-30T17:18:03Z</time> <lat>60.42339</lat> <ele>490</ele> </tp>
 <tp> <time>2006-04-30T17:18:05Z</time> <lat>60.42338</lat> <lon>5.34269</lon> <ele>487</ele> </tp>
 </track>
 </gpsml>
@@ -872,18 +874,19 @@
 
 TODO: {
     local $TODO = "Shall lat/lon be cleared if one is missing?";
+    diag("Testing --require...");
     testcmd("../gpst -re missing.gpsml", # {{{
         <<END,
 <?xml version="1.0" encoding="UTF-8"?>
 <gpsml>
 <track>
 <title>Missing various elements</title>
-<tp> <time>2006-04-30T17:17:11Z</time> <ele>483</ele> </tp>
+<tp> <time>2006-04-30T17:17:11Z</time> <lon>5.34187</lon> <ele>483</ele> </tp>
 <tp> <time>2006-04-30T17:17:22Z</time> <ele>485</ele> </tp>
 <tp> <lat>60.42347</lat> <lon>5.34212</lon> <ele>486</ele> </tp>
-<tp> <ele>484</ele> </tp>
+<tp> <lon>5.34224</lon> <ele>484</ele> </tp>
 <tp> <ele>486</ele> </tp>
-<tp> <time>2006-04-30T17:18:03Z</time> <ele>490</ele> </tp>
+<tp> <time>2006-04-30T17:18:03Z</time> <lat>60.42339</lat> <ele>490</ele> </tp>
 <tp> <time>2006-04-30T17:18:05Z</time> <lat>60.42338</lat> <lon>5.34269</lon> <ele>487</ele> </tp>
 </track>
 </gpsml>
@@ -899,11 +902,11 @@
 <track>
 <title>Missing various elements</title>
 <tp> <time>2006-04-30T17:16:59Z</time> </tp>
-<tp> <time>2006-04-30T17:17:00Z</time> </tp>
+<tp> <time>2006-04-30T17:17:00Z</time> <lat>60.42352</lat> </tp>
 <tp> <time>2006-04-30T17:17:09Z</time> <lat>60.42353</lat> <lon>5.34185</lon> </tp>
-<tp> <time>2006-04-30T17:17:11Z</time> <ele>483</ele> </tp>
+<tp> <time>2006-04-30T17:17:11Z</time> <lon>5.34187</lon> <ele>483</ele> </tp>
 <tp> <time>2006-04-30T17:17:22Z</time> <ele>485</ele> </tp>
-<tp> <time>2006-04-30T17:18:03Z</time> <ele>490</ele> </tp>
+<tp> <time>2006-04-30T17:18:03Z</time> <lat>60.42339</lat> <ele>490</ele> </tp>
 <tp> <time>2006-04-30T17:18:05Z</time> <lat>60.42338</lat> <lon>5.34269</lon> <ele>487</ele> </tp>
 </track>
 </gpsml>
@@ -918,8 +921,14 @@
 <gpsml>
 <track>
 <title>Missing various elements</title>
+<tp> <time>2006-04-30T17:17:00Z</time> <lat>60.42352</lat> </tp>
 <tp> <time>2006-04-30T17:17:09Z</time> <lat>60.42353</lat> <lon>5.34185</lon> </tp>
+<tp> <time>2006-04-30T17:17:11Z</time> <lon>5.34187</lon> <ele>483</ele> </tp>
 <tp> <lat>60.42347</lat> <lon>5.34212</lon> <ele>486</ele> </tp>
+<tp> <lon>5.34224</lon> <ele>484</ele> </tp>
+<tp> <lat>60.42339</lat> </tp>
+<tp> <lon>5.34262</lon> </tp>
+<tp> <time>2006-04-30T17:18:03Z</time> <lat>60.42339</lat> <ele>490</ele> </tp>
 <tp> <time>2006-04-30T17:18:05Z</time> <lat>60.42338</lat> <lon>5.34269</lon> <ele>487</ele> </tp>
 </track>
 </gpsml>
@@ -934,9 +943,9 @@
 <gpsml>
 <track>
 <title>Missing various elements</title>
-<tp> <time>2006-04-30T17:17:11Z</time> <ele>483</ele> </tp>
+<tp> <time>2006-04-30T17:17:11Z</time> <lon>5.34187</lon> <ele>483</ele> </tp>
 <tp> <time>2006-04-30T17:17:22Z</time> <ele>485</ele> </tp>
-<tp> <time>2006-04-30T17:18:03Z</time> <ele>490</ele> </tp>
+<tp> <time>2006-04-30T17:18:03Z</time> <lat>60.42339</lat> <ele>490</ele> </tp>
 <tp> <time>2006-04-30T17:18:05Z</time> <lat>60.42338</lat> <lon>5.34269</lon> <ele>487</ele> </tp>
 </track>
 </gpsml>
@@ -951,6 +960,15 @@
 <gpsml>
 <track>
 <title>Missing various elements</title>
+<tp> <time>2006-04-30T17:16:59Z</time> </tp>
+<tp> <time>2006-04-30T17:17:00Z</time> <lat>60.42352</lat> </tp>
+<tp> <time>2006-04-30T17:17:09Z</time> <lat>60.42353</lat> <lon>5.34185</lon> </tp>
+<tp> <time>2006-04-30T17:17:11Z</time> <lon>5.34187</lon> <ele>483</ele> </tp>
+<tp> <time>2006-04-30T17:17:22Z</time> <ele>485</ele> </tp>
+<tp> <lat>60.42347</lat> <lon>5.34212</lon> <ele>486</ele> </tp>
+<tp> <lon>5.34224</lon> <ele>484</ele> </tp>
+<tp> <ele>486</ele> </tp>
+<tp> <time>2006-04-30T17:18:03Z</time> <lat>60.42339</lat> <ele>490</ele> </tp>
 <tp> <time>2006-04-30T17:18:05Z</time> <lat>60.42338</lat> <lon>5.34269</lon> <ele>487</ele> </tp>
 </track>
 </gpsml>
@@ -965,7 +983,14 @@
 <gpsml>
 <track>
 <title>Missing various elements</title>
+<tp> <time>2006-04-30T17:17:00Z</time> <lat>60.42352</lat> </tp>
+<tp> <time>2006-04-30T17:17:09Z</time> <lat>60.42353</lat> <lon>5.34185</lon> </tp>
+<tp> <time>2006-04-30T17:17:11Z</time> <lon>5.34187</lon> <ele>483</ele> </tp>
+<tp> <time>2006-04-30T17:17:22Z</time> <ele>485</ele> </tp>
 <tp> <lat>60.42347</lat> <lon>5.34212</lon> <ele>486</ele> </tp>
+<tp> <lon>5.34224</lon> <ele>484</ele> </tp>
+<tp> <ele>486</ele> </tp>
+<tp> <time>2006-04-30T17:18:03Z</time> <lat>60.42339</lat> <ele>490</ele> </tp>
 <tp> <time>2006-04-30T17:18:05Z</time> <lat>60.42338</lat> <lon>5.34269</lon> <ele>487</ele> </tp>
 </track>
 </gpsml>
@@ -975,37 +1000,85 @@
 
     # }}}
 }
+    testcmd("../gpst -o gpx missing.gpsml | ../gpst", # {{{
+        <<END,
+<?xml version="1.0" encoding="UTF-8"?>
+<gpsml>
+<track>
+<title>Missing various elements</title>
+<tp> <time>2006-04-30T17:17:09Z</time> <lat>60.42353</lat> <lon>5.34185</lon> </tp>
+<tp> <lat>60.42347</lat> <lon>5.34212</lon> <ele>486</ele> </tp>
+<tp> <time>2006-04-30T17:18:05Z</time> <lat>60.42338</lat> <lon>5.34269</lon> <ele>487</ele> </tp>
+</track>
+</gpsml>
+END
+    "Via GPX from gpsml with missing data",
+    );
 
-testcmd("../gpst -o gpx missing.gpsml", # {{{
+# }}}
+testcmd("echo '<tp> </tp>' | ../gpst", # {{{
     <<END,
+<?xml version="1.0" encoding="UTF-8"?>
+<gpsml>
+<track>
+</track>
+</gpsml>
+END
+    "Don’t print empty trackpoints"
+);
+
+# }}}
+testcmd("echo '<tp> <lat>5</lat> </tp>' | ../gpst", # {{{
+    <<END,
+<?xml version="1.0" encoding="UTF-8"?>
+<gpsml>
+<track>
+<etp err="incomplete"> <lat>5</lat> </etp>
+</track>
+</gpsml>
+END
+    "Only latitude, no longitude (gpsml)"
+);
+
+# }}}
+testcmd("echo '<tp> <lon>5</lon> </tp>' | ../gpst", # {{{
+    <<END,
+<?xml version="1.0" encoding="UTF-8"?>
+<gpsml>
+<track>
+<etp err="incomplete"> <lon>5</lon> </etp>
+</track>
+</gpsml>
+END
+    "Only longitude, no latitude (gpsml)"
+);
+
+# }}}
+testcmd("echo '<tp> <lat>5</lat> </tp>' | ../gpst -o gpx", # {{{
+    <<END,
 $gpx_header
   <trk>
     <trkseg>
-      <trkpt lat="60.42353" lon="5.34185"> <time>2006-04-30T17:17:09Z</time> </trkpt>
-      <trkpt> <time>2006-04-30T17:17:11Z</time> <ele>483</ele> </trkpt>
-      <trkpt> <time>2006-04-30T17:17:22Z</time> <ele>485</ele> </trkpt>
-      <trkpt lat="60.42347" lon="5.34212"> <ele>486</ele> </trkpt>
-      <trkpt> <ele>484</ele> </trkpt>
-      <trkpt> <ele>486</ele> </trkpt>
-      <trkpt> <time>2006-04-30T17:18:03Z</time> <ele>490</ele> </trkpt>
-      <trkpt lat="60.42338" lon="5.34269"> <time>2006-04-30T17:18:05Z</time> <ele>487</ele> </trkpt>
     </trkseg>
   </trk>
 </gpx>
 END
-    "Output GPX from gpsml with missing data",
-    );
+    "Only latitude, no longitude (gpx)"
+);
 
 # }}}
-testcmd("echo '<tp> </tp>' | ../gpst", # {{{
+testcmd("echo '<tp> <lon>5</lon> </tp>' | ../gpst -o gpx", # {{{
     <<END,
-<?xml version="1.0" encoding="UTF-8"?>
-<gpsml>
-<track>
-</track>
-</gpsml>
+<?xml version="1.0" standalone="no"?>
+<gpx>
+  <trk>
+    <trkseg>
+    </trkseg>
+  </trk>
+</gpx>
 END
-    "Don’t print empty trackpoints");
+    "Only longitude, no latitude (gpx)"
+);
 
 # }}}
 testcmd("../gpst --epoch pause.gpx", # {{{
Index: GPST.pm
===================================================================
--- GPST.pm	(revision 1843)
+++ GPST.pm	(working copy)
@@ -104,26 +104,23 @@
             # }}}
         } elsif($Dat{'format'} eq "gpx") {
             # {{{
-            my $lat_str = length($Dat{'lat'}) ? " lat=\"$Dat{'lat'}\"" : "";
-            my $lon_str = length($Dat{'lon'}) ? " lon=\"$Dat{'lon'}\"" : "";
-            if (length("$lat_str$lon_str$Dat{'ele'}")) {
-                $Retval .=
-                join("",
-                    "$Spc$Spc$Spc$Spc$Spc$Spc",
-                    "<trkpt$lat_str$lon_str>",
-                    "$Spc",
-                    $print_time
-                        ? "<time>" .
-                          "$Dat{'year'}-$Dat{'month'}-$Dat{'day'}T" .
-                          "$Dat{'hour'}:$Dat{'min'}:$Dat{'sec'}Z" .
-                          "</time>$Spc"
-                        : "",
-                    length($Dat{'ele'})
-                        ? "<ele>$Dat{'ele'}</ele>$Spc"
-                        : "",
-                    "</trkpt>\n"
-                );
-            }
+            (length($Dat{'lat'}) && length($Dat{'lon'})) || return("");
+            $Retval .=
+            join("",
+                "$Spc$Spc$Spc$Spc$Spc$Spc",
+                "<trkpt lat=\"$Dat{'lat'}\" lon=\"$Dat{'lon'}\">",
+                "$Spc",
+                $print_time
+                    ? "<time>" .
+                      "$Dat{'year'}-$Dat{'month'}-$Dat{'day'}T" .
+                      "$Dat{'hour'}:$Dat{'min'}:$Dat{'sec'}Z" .
+                      "</time>$Spc"
+                    : "",
+                length($Dat{'ele'})
+                    ? "<ele>$Dat{'ele'}</ele>$Spc"
+                    : "",
+                "</trkpt>\n"
+            );
             # }}}
         } else {
             $Retval = undef;
Index: gpst
===================================================================
--- gpst	(revision 1843)
+++ gpst	(working copy)
@@ -710,9 +710,14 @@
     defined($Dat{'lon'}) || ($Dat{'lon'} = "");
     defined($Dat{'year'}) || ($Dat{'year'} = "");
     my $print_time = length($Dat{'year'}) ? 1 : 0;
-    my $print_pos = (length($Dat{'lat'}) && length($Dat{'lon'})) ? 1 : 0;
-    if (!$print_pos) {
-        $Dat{'lat'} = $Dat{'lon'} = "";
+    my $print_pos = 1;
+    if (length($Dat{'lat'}) && length($Dat{'lon'})) {
+        $print_pos = 1;
+    } else {
+        if (length($Dat{'lat'}) || length($Dat{'lon'})) {
+            $Dat{'error'} = "incomplete";
+            $print_pos = ($Opt{'output-format'} eq "gpsml") ? 1 : 0;
+        }
     }
     my $print_ele = length($Dat{'ele'}) ? 1 : 0;
     my $print_desc = length($Dat{'desc'}) ? 1 : 0;
@@ -829,11 +834,9 @@
 
     if ($Dat{'type'} eq "tp") {
         # {{{
-        if ($Opt{'require'}) {
-            $Req{'time'} && !$print_time && return;
-            $Req{'position'} && !$print_pos && return;
-            $Req{'ele'} && !$print_ele && return;
-        }
+        ($Req{'time'} && !$print_time) && return("");
+        ($Req{'position'} && length("$Dat{'lat'}$Dat{'lon'}")) && return("");
+        ($Req{'ele'} && !$print_ele) && return("");
 
         if ($Opt{'inside'} || $Opt{'outside'}) {
             if (
