Index: GPST.pm
===================================================================
--- GPST.pm	(revision 1830)
+++ GPST.pm	(working copy)
@@ -36,6 +36,7 @@
     defined($Dat{'type'}) || return(undef);
     defined($Dat{'format'}) || return(undef);
     defined($Dat{'error'}) || return(undef);
+    defined($Dat{'break'}) || ($Dat{'break'} = 0);
 
     defined($Dat{'year'}) || ($Dat{'year'} = 0);
     defined($Dat{'month'}) || ($Dat{'month'} = 0);
@@ -72,7 +73,9 @@
         if ($Dat{'format'} eq "gpsml") {
             # {{{
             my $Elem = length($err_str) ? "etp" : "tp";
-            $Retval .= join("",
+            my $tp_str = "";
+            $Dat{'break'} && ($Retval .= "<break/>\n");
+            $tp_str .= join("",
                     $print_time
                         ? sprintf("<time>%04u-%02u-%02uT" .
                                   "%02u:%02u:%02gZ</time> ",
@@ -94,20 +97,30 @@
                                   $Dat{'desc'})
                         : ""
             );
-            length($Retval) &&
-                ($Retval = sprintf("<%s%s> %s</%s>\n",
+            if (length($tp_str)) {
+                ($tp_str = sprintf("<%s%s> %s</%s>\n",
                                    $Elem,
                                    length($err_str) ? " err=\"$err_str\"" : "",
-                                   $Retval,
-                                   $Elem)
-            );
+                                   $tp_str,
+                                   $Elem
+                           )
+                );
+            }
+            $Retval .= $tp_str;
+            $Dat{'break'} = 0;
             # }}}
+            # }}}
         } elsif($Dat{'format'} eq "gpx") {
             # {{{
+            my $tp_str = "";
             my $lat_str = length($Dat{'lat'}) ? " lat=\"$Dat{'lat'}\"" : "";
             my $lon_str = length($Dat{'lon'}) ? " lon=\"$Dat{'lon'}\"" : "";
+            if ($Dat{'break'}) {
+                $tp_str .= "$Spc$Spc$Spc$Spc</trkseg>\n" .
+                           "$Spc$Spc$Spc$Spc<trkseg>\n";
+            }
             if (length("$lat_str$lon_str$Dat{'ele'}")) {
-                $Retval .=
+                $tp_str .=
                 join("",
                     "$Spc$Spc$Spc$Spc$Spc$Spc",
                     "<trkpt$lat_str$lon_str>",
@@ -124,6 +137,7 @@
                     "</trkpt>\n"
                 );
             }
+            $Retval .= $tp_str;
             # }}}
         } else {
             $Retval = undef;
Index: gpst
===================================================================
--- gpst	(revision 1830)
+++ gpst	(working copy)
@@ -100,7 +100,6 @@
 my $DIGIT = '[0-9\.\-\+]'; # Used in regexps
 $GPST::Spc = $Opt{'strip-whitespace'} ? "" : " ";
 my $Spc = $GPST::Spc; # FIXME
-my $found_move = 0; # Settes til 1 hvis en /^# move$/ blir funnet.
 my $first_time = 0;
 my $last_time = 0;
 my ($last_lon, $last_lat, $last_ele, $last_line) =
@@ -294,7 +293,7 @@
                 print_entry(%Dat);
                 # }}}
             } elsif (m#^<break\b.*?/>#) {
-                $found_move = 1;
+                $Dat{'break'} = 1;
             } elsif (m#^<(title|pause)\b.*?>(.*?)</(title|pause)>#) {
                 $Dat{'type'} = $1;
                 $Dat{$1} = $2;
@@ -318,7 +317,7 @@
                 read_xmlfile($xml_data);
                 last;
             } elsif (/^move$/) {
-                $found_move = 1;
+                $Dat{'break'} = 1;
             } elsif (m#^(\d+)\t($DIGIT+)\t($DIGIT+)\t($DIGIT)#) {
                 # CSV format, epoch style {{{
                 my ($ep_time, $lon_val, $lat_val, $Alt) =
@@ -405,7 +404,7 @@
             } elsif (/^Track\t(.*?)\t/) {
                 $Dat{'title'} = txt_to_xml($1);
                 $Dat{'type'} = "title";
-                $found_move = 1;
+                $Dat{'break'} = 1;
                 print_entry(%Dat);
             } elsif (
                 m{^
@@ -517,13 +516,9 @@
                 print_entry(%Dat);
                 # }}}
             } elsif (/^xmaplog /) {
-                ($Opt{'output-format'} eq "csv")
-                    && ($Opt{'save-to-file'} eq "\n")
-                    && print("\n");
+                $Dat{'break'} = 1;
             } elsif (/^$/) {
-                ($Opt{'output-format'} eq "csv")
-                    && ($Opt{'save-to-file'} eq "\n")
-                    && print("\n");
+                ($Opt{'output-format'} eq "csv") && ($Dat{'break'} = 1);
             } elsif (/^Pause: /) {
                 # NOP, is here to cope with old files I’ve lying around.
             } elsif ($Dat{'error'} eq "desc") {
@@ -585,6 +580,7 @@
         }
         {
             my $el_trkseg = $2;
+            $Dat{'break'} = 1;
             $el_trkseg =~
             s{
                 <trkpt\b(.*?)>(.*?)</trkpt>
@@ -618,9 +614,9 @@
                 print_entry(%Dat);
                 "";
             }gsex;
-            $found_move = 1;
+            $Dat{'break'} = 1;
         }gsex;
-        $found_move = 1;
+        $Dat{'break'} = 1;
     }gsex;
     # }}}
 }
@@ -756,6 +752,7 @@
                 # FIXME: Make --fix work with gpx.
                 if ($Opt{'fix'} && ($Opt{'output-format'} !~ /^gpx$/)) {
                     $Dat{'error'} = "chrono";
+                    $Dat{'break'} = 1;
                 }
             }
         }
@@ -972,7 +969,7 @@
     }
 
     if ($do_print) {
-        if ($found_move) {
+        if ($Dat{'break'}) {
             if ($Opt{'output-format'} eq "gpsml") {
                 $Line = "<break/>\n$Line";
             }
@@ -983,7 +980,7 @@
                 $Line .= "$Spc$Spc$Spc$Spc</trkseg>\n" .
                          "$Spc$Spc$Spc$Spc<trkseg>\n";
             }
-            $found_move = 0;
+            $Dat{'break'} = 0;
         }
         print($Line);
     }
