$Id$

Index: src/ops.c
===================================================================
RCS file: /cvsroot/vim/vim/src/ops.c,v
retrieving revision 1.109
diff -u -r1.109 ops.c
--- src/ops.c	13 Dec 2004 06:34:59 -0000	1.109
+++ src/ops.c	3 Jul 2005 11:39:35 -0000
@@ -6026,12 +6026,13 @@
 	    col_print(buf2, (int)STRLEN(p), linetabsize(p));
 
 	    sprintf((char *)IObuff,
-		_("Col %s of %s; Line %ld of %ld; Word %ld of %ld; Byte %ld of %ld"),
+		_("Col %s of %s; Line %ld of %ld; Word %ld of %ld; Byte %ld of %ld (%.4f%%)"),
 		    (char *)buf1, (char *)buf2,
 		    (long)curwin->w_cursor.lnum,
 		    (long)curbuf->b_ml.ml_line_count,
 		    word_count_cursor, word_count,
-		    char_count_cursor, char_count);
+		    char_count_cursor, char_count,
+		    char_count ? (1.0*char_count_cursor)/(1.0*char_count)*100.0 : 100);
 	}
 
 #ifdef FEAT_MBYTE
