Index: wlan-list
===================================================================
--- wlan-list	(revision 2022)
+++ wlan-list	(working copy)
@@ -49,10 +49,12 @@
 if ($Opt{'verbose'}) {
     system("iwlist ath0 scan | less");
 } elsif ($Opt{'encrypted'}) {
-    system("iwlist ath0 scan | egrep '(Address:|ESSID:|Quality=|Encryption)' | perl -pe 's/(Encryption key:off)/\x1B[1m$1\x1B[m/g;'");
+    # system("iwlist ath0 scan | egrep '(Address:|ESSID:|Quality=|Encryption)' | perl -pe 's/(Encryption key:off)/\x1B[1m$1\x1B[m/g;'");
+    system("cat ~/bin/exemp | egrep '(Address:|ESSID:|Quality=|Encryption)' | perl -pe 's/(Encryption key:off)/\x1B[1m$1\x1B[m/g;'");
 } elsif ($Opt{'xml'}) {
     my $found_cell = 0;
-    if (open(PipeFP, "iwlist ath0 scan |")) {
+    # if (open(PipeFP, "iwlist ath0 scan |")) {
+    if (open(PipeFP, "cat ~/bin/exemp |")) {
         print("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<wlan>\n");
         my $curr_time = time();
         my ($Sec, $Min, $Hour, $Day, $Mon, $Year, $Wday, $Yday) = gmtime($curr_time);
Index: view_df
===================================================================
--- view_df	(revision 2022)
+++ view_df	(working copy)
@@ -12,9 +12,9 @@
 fi
 
 if [ "$2" != "" ]; then
-    grep $1 $lfile | tr '\t' ' ' | tr -s ' ' | cut -d ' ' -f 1,5 | tail -$2 >~/log/data
+    zgrep $1 $lfile | tr '\t' ' ' | tr -s ' ' | cut -d ' ' -f 1,5 | tail -$2 >~/log/data
 else
-    grep $1 $lfile | tr '\t' ' ' | tr -s ' ' | cut -d ' ' -f 1,5 >~/log/data
+    zgrep $1 $lfile | tr '\t' ' ' | tr -s ' ' | cut -d ' ' -f 1,5 >~/log/data
 fi
 
 cd ~/log
Index: Lib/std/db-catalog
===================================================================
--- Lib/std/db-catalog	(revision 2022)
+++ Lib/std/db-catalog	(working copy)
@@ -1,44 +1,12 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!DOCTYPE catalog
-   PUBLIC "-//OASIS/DTD Entity Resolution XML Catalog V1.0//EN"
-   "http://www.oasis-open.org/committees/entity/release/1.0/catalog.dtd">
-
+<!DOCTYPE catalog PUBLIC "-//OASIS//DTD Entity Resolution XML Catalog V1.0//EN" "http://www.oasis-open.org/committees/entity/release/1.0/catalog.dtd">
 <catalog xmlns="urn:oasis:names:tc:entity:xmlns:xml:catalog">
+  <!-- $Id$ -->
+  <public publicId="-//OASIS//DTD DocBook XML V4//EN"
+    uri="/usr/share/xml/docbook/schema/dtd/4.3/docbookx.dtd"
+  />
 
-  <!-- DTD and stylesheet files installed under /usr/share/xml -->
-  <group xml:base="file:///usr/share/xml/" >
-
-    <!-- Resolve DTD URL system ID to local file -->
-    <rewriteSystem
-        systemIdStartString="http://www.oasis-open.org/docbook/xml/4.4/"
-        rewritePrefix="docbook44/" />
-    <!-- Resolve stylesheet URL to local file -->
-    <rewriteURI
-        uriStartString="http://docbook.sourceforge.net/release/xsl/current/"
-        rewritePrefix="docbook-xsl-1.68.1/" />
-
-    <!-- Resolve DTD PUBLIC identifiers -->
-    <nextCatalog  catalog="docbook44/catalog.xml" />
-
-    <!-- To resolve simple DTD SYSTEM identifiers. -->
-    <!-- Note: this does not work with Java resolver -->
-    <!--   classes in Saxon or Xalan -->
-    <system
-         systemId="docbook.dtd"
-         uri="docbook44/docbookx.dtd" />
-
-    <!-- To resolve short stylesheet references -->
-    <uri
-        name="docbook.xsl"
-        uri="docbook-xsl-1.68.1/html/docbook.xsl" />
-    <uri
-        name="chunk.xsl"
-        uri="docbook-xsl-1.68.1/html/chunk.xsl" />
-    <uri
-        name="fo-docbook.xsl"
-        uri="docbook-xsl-1.68.1/fo/docbook.xsl" />
-
-  </group>
-
+  <public publicId="-//Norman Walsh//DTD Slides V3.4//EN"
+    uri="/home/sunny/src/other/docbook/docbook-slides-3.4.0/schema/dtd/slides.dtd"
+  />
 </catalog>
-
