Index: update_things.sql
===================================================================
--- update_things.sql	(revision 800)
+++ update_things.sql	(working copy)
@@ -1,17 +1,20 @@
 -- $Id$
 
 -- OBS! Må fjernes når jeg er ferdig med å teste og opprenskinga er gjort.
-\echo Slett skrotpunkter.
-DELETE FROM logg WHERE lat < 51;
-DELETE FROM logg WHERE lat > 71;
-DELETE FROM logg WHERE lon < -2;
-DELETE FROM logg WHERE lon > 26;
-DELETE FROM logg WHERE date < '2002-1-1';
-DELETE FROM logg WHERE date > '2007-1-1';
-DELETE FROM logg WHERE date BETWEEN '2005-9-24' AND '2006-2-8';
-DELETE FROM logg WHERE date BETWEEN '2003-02-15 17:58:26Z' AND '2003-02-15 17:59:37Z';
-DELETE FROM logg WHERE date BETWEEN '2003-07-15 16:06:58Z' AND '2003-07-15 16:08:05Z';
-DELETE FROM logg WHERE alt = -1500;
+-- \echo Slett skrotpunkter.
+SELECT * FROM logg WHERE lat < 51 ORDER BY date;
+SELECT * FROM logg WHERE lat > 71 ORDER BY date;
+SELECT * FROM logg WHERE lon < -2 ORDER BY date;
+SELECT * FROM logg WHERE lon > 26 ORDER BY date;
+SELECT * FROM logg WHERE date < '2002-1-1' ORDER BY date;
+SELECT * FROM logg WHERE date > '2007-1-1' ORDER BY date;
+SELECT * FROM logg WHERE date BETWEEN '2005-9-24' AND '2006-2-8' ORDER BY date;
+SELECT * FROM logg WHERE date BETWEEN '2003-02-15 17:58:26Z' AND '2003-02-15 17:59:37Z' ORDER BY date;
+SELECT * FROM logg WHERE date BETWEEN '2003-07-15 16:06:58Z' AND '2003-07-15 16:08:05Z' ORDER BY date;
+SELECT * FROM logg WHERE date = '2002-10-23 13:06:55Z' ORDER BY date;
+SELECT * FROM logg WHERE date = '2003-05-29 16:26:00Z' ORDER BY date;
+SELECT * FROM logg WHERE date = '2003-01-05 18:33:10Z' ORDER BY date;
+SELECT * FROM logg WHERE alt = -1500 ORDER BY date;
 \echo
 \echo Oppdater koor
 UPDATE logg SET koor = point(lat,lon) WHERE koor IS NULL;
