Index: postgres/create_table.sql
===================================================================
--- postgres/create_table.sql	(revisjon 2455)
+++ postgres/create_table.sql	(arbeidskopi)
@@ -46,6 +46,17 @@
     author text
 );
 
+CREATE TABLE files (
+    mtime timestamptz,
+    coor point,
+    filename text,
+    descr text,
+    size bigint,
+    sha1 varchar(40),
+    md5 varchar(40),
+    path text
+);
+
 CREATE TABLE stat (
     lastupdate timestamptz,
     laststed timestamptz
Index: postgres/create_views.sql
===================================================================
--- postgres/create_views.sql	(revisjon 2455)
+++ postgres/create_views.sql	(arbeidskopi)
@@ -133,6 +133,22 @@
     ) AS u
     ORDER BY date;
 
+CREATE OR REPLACE VIEW evfiles AS
+    SELECT * FROM (
+        SELECT     'gps' AS flag, date, coor, sted || ' (' || dist || ')' AS sted, ele::numeric(8,1), NULL AS descr, avst
+            FROM logg
+        UNION ALL
+        SELECT   'event' AS flag, date, coor, NULL, NULL, descr AS descr, NULL
+            FROM events
+        UNION ALL
+        SELECT     'pic' AS flag, date, coor, filename, NULL, NULL, NULL
+            FROM pictures
+        UNION ALL
+        SELECT     'file' AS flag, mtime, coor, filename, NULL, path, NULL
+            FROM files
+    ) AS u
+    ORDER BY date;
+
 CREATE OR REPLACE VIEW wp AS
     SELECT
         coor AS coor,
Index: BRANCHINFO
===================================================================
--- BRANCHINFO	(revisjon 0)
+++ BRANCHINFO	(revisjon 0)
@@ -0,0 +1,6 @@
+$Id$
+
+Legger inn informasjon om filer i Postgres. Fildato, navn, plassering og 
+så videre skal kunne mikses inn med alt det andre.
+
+vim: set tw=72 et fo=tcqw :
