Index: src/feature.h
===================================================================
RCS file: /cvsroot/vim/vim/src/feature.h,v
retrieving revision 1.60
diff -u -r1.60 feature.h
--- src/feature.h	8 May 2004 00:02:51 -0000	1.60
+++ src/feature.h	26 Sep 2004 09:34:41 -0000
@@ -913,7 +913,7 @@
  *			a modifed version of Vim.
  *			Also from the "--with-modified-by" configure argument.
  */
-/* #define MODIFIED_BY "John Doe" */
+#define MODIFIED_BY "Øyvind A. Holm <sunny@sunbase.org>"
 
 /*
  * Machine dependent:
