$Id$

--- src/orig.ops.c	2004-12-20 06:16:13.000000000 +0100
+++ src/ops.c	2005-02-07 11:39:45.000000000 +0100
@@ -6026,12 +6026,13 @@
 	    col_print(buf2, (int)STRLEN(p), linetabsize(p));
 
 	    sprintf((char *)IObuff,
-		_("Col %s of %s; Line %ld of %ld; Word %ld of %ld; Byte %ld of %ld"),
+		_("Col %s of %s; Line %ld of %ld; Word %ld of %ld; Byte %ld of %ld (%.4f%%)"),
 		    (char *)buf1, (char *)buf2,
 		    (long)curwin->w_cursor.lnum,
 		    (long)curbuf->b_ml.ml_line_count,
 		    word_count_cursor, word_count,
-		    char_count_cursor, char_count);
+		    char_count_cursor, char_count,
+		    char_count ? (1.0*char_count_cursor)/(1.0*char_count)*100.0 : 100);
 	}
 
 #ifdef FEAT_MBYTE
