$Id$

Index: src/feature.h
===================================================================
RCS file: /cvsroot/vim/vim7/src/feature.h,v
retrieving revision 1.15
diff -u -r1.15 feature.h
--- src/feature.h	30 Jun 2005 21:56:01 -0000	1.15
+++ src/feature.h	3 Jul 2005 11:23:14 -0000
@@ -924,7 +924,7 @@
  *			a modifed version of Vim.
  *			Also from the "--with-modified-by" configure argument.
  */
-/* #define MODIFIED_BY "John Doe" */
+#define MODIFIED_BY "Øyvind A. Holm <sunny@sunbase.org>"
 
 /*
  * Machine dependent:
