# File ID: 78d9ee7a-0b5e-11de-948f-000475e441b9
Index: finduuid
===================================================================
--- finduuid	(revisjon 3062)
+++ finduuid	(arbeidskopi)
@@ -26,6 +26,7 @@
 
     'debug' => 0,
     'filenames' => 0,
+    'grep' => 0,
     'help' => 0,
     'line' => 0,
     'verbose' => 0,
@@ -47,6 +48,7 @@
 
     "debug" => \$Opt{'debug'},
     "filenames|f" => \$Opt{'filenames'},
+    "grep|g" => \$Opt{'grep'},
     "help|h" => \$Opt{'help'},
     "line|l" => \$Opt{'line'},
     "verbose|v+" => \$Opt{'verbose'},
@@ -66,11 +68,16 @@
 
 while (<>) {
     chomp(my $Line = $_);
-    if (/($Templ)/) {
-        if ($Opt{'line'}) {
-            printf("%s%s\n", ($Opt{'filenames'} ? "$ARGV:" : ""), $Line);
-        } else {
-            $Line =~ s/($Templ)/printf("%s%s\n", ($Opt{'filenames'} ? "$ARGV:" : ""), $1)/ge;
+    if ($Opt{'grep'}) {
+        s/($Templ)/`grep -h ' u="$1"' ~\/uuids\/0* | grep '<suuid ' | head -1 | xf | grep -v '<?xml version=' | perl -e 'my \$a = join("", <STDIN>); \$a =~ s\/\n\$\/\/s; print \$a;'`/ge;
+        print;
+    } else {
+        if (/($Templ)/) {
+            if ($Opt{'line'}) {
+                printf("%s%s\n", ($Opt{'filenames'} ? "$ARGV:" : ""), $Line);
+            } else {
+                $Line =~ s/($Templ)/printf("%s%s\n", ($Opt{'filenames'} ? "$ARGV:" : ""), $1)/ge;
+            }
         }
     }
 }
