Index: svnlog2tab
===================================================================
--- svnlog2tab	(revision 2896)
+++ svnlog2tab	(working copy)
@@ -16,7 +16,10 @@
 }
 
 use strict;
+# use diagnostics;
 use Getopt::Long;
+use XML::Simple;
+use Data::Dumper;
 
 $| = 1;
 
@@ -57,10 +60,23 @@
     exit(0);
 }
 
-while (<>) {
-    print;
-}
+my $Orig = join("", <>);
+# print("orig = '$Orig'\n");
 
+my $Data = XMLin($Orig, ForceArray => 1);
+
+print(Dumper($Data));
+exit;
+my $i = 0;
+
+print(
+    join("\t",
+        $Data->{logentry}->{author}[$i],
+        $Data->{logentry}->{date}[$i]
+    ) . "\n"
+);
+
+
 sub print_version {
     # Print program version {{{
     for (@main::version_array) {
