$Id$

Index: src/feature.h
===================================================================
RCS file: /cvsroot/vim/vim7/src/feature.h,v
retrieving revision 1.38
diff -u -r1.38 feature.h
--- src/feature.h	6 Aug 2008 16:45:07 -0000	1.38
+++ src/feature.h	30 Aug 2008 08:25:33 -0000
@@ -967,7 +967,7 @@
  *			a modifed version of Vim.
  *			Also from the "--with-modified-by" configure argument.
  */
-/* #define MODIFIED_BY "John Doe" */
+#define MODIFIED_BY "Øyvind A. Holm <sunny@sunbase.org>"
 
 /*
  * Machine dependent:
