Index: create_funcs.sql
===================================================================
--- create_funcs.sql	(revision 3096)
+++ create_funcs.sql	(working copy)
@@ -31,6 +31,15 @@
     ) AS s;
 $$ LANGUAGE SQL; -- }}}
 
+-- closelog(): Lister ut de nærmeste punktene i loggen for et koordinat
+CREATE OR REPLACE FUNCTION closelog(point) RETURNS void -- {{{
+AS $$
+BEGIN
+    PERFORM (point($1) <-> coor)::numeric(8,5) AS distance, *
+        FROM logg ORDER BY distance LIMIT 100;
+END;
+$$ LANGUAGE plpgsql; -- }}}
+
 -- findpos(): Beregn koordinater for et tidspunkt som ligger mellom to trackpunkter.
 CREATE OR REPLACE FUNCTION findpos(currtime timestamptz) RETURNS point AS $$ -- {{{
 DECLARE
