Index: tests/run-tests.pl
===================================================================
--- tests/run-tests.pl	(revision 2020)
+++ tests/run-tests.pl	(working copy)
@@ -488,9 +488,11 @@
       <trkpt lat="70.6800486" lon="23.6746151"> <time>2002-12-22T21:42:24Z</time> </trkpt>
       <trkpt lat="70.6799322" lon="23.6740038"> <time>2002-12-22T21:42:32Z</time> </trkpt>
       <trkpt lat="70.6796266" lon="23.6723991"> <time>2002-12-22T21:42:54Z</time> </trkpt>
+      <!-- 20021222T214351-20021222T214354: CO: No signal \x7B\x7B\x7B -->
       <!-- <trkpt lat="70.6796266" lon="23.6723991"> <time>2002-12-22T21:43:51Z</time> <extensions> <error>desc</error> </extensions> </trkpt> -->
       <!-- <trkpt lat="70.6796266" lon="23.6723991"> <time>2002-12-22T21:43:52Z</time> <extensions> <error>desc</error> </extensions> </trkpt> -->
       <!-- <trkpt lat="70.6796266" lon="23.6723991"> <time>2002-12-22T21:43:54Z</time> <extensions> <error>desc</error> </extensions> </trkpt> -->
+      <!-- 20021222T214351-20021222T214354: CO: No signal \x7D\x7D\x7D -->
       <trkpt lat="70.6800774" lon="23.6757566"> <time>2002-12-22T21:44:45Z</time> </trkpt>
     </trkseg>
     <trkseg>
@@ -512,9 +514,11 @@
 <trkpt lat="70.6800486" lon="23.6746151"><time>2002-12-22T21:42:24Z</time></trkpt>
 <trkpt lat="70.6799322" lon="23.6740038"><time>2002-12-22T21:42:32Z</time></trkpt>
 <trkpt lat="70.6796266" lon="23.6723991"><time>2002-12-22T21:42:54Z</time></trkpt>
+<!-- 20021222T214351-20021222T214354: CO: No signal \x7B\x7B\x7B -->
 <!-- <trkpt lat="70.6796266" lon="23.6723991"><time>2002-12-22T21:43:51Z</time><extensions><error>desc</error></extensions></trkpt> -->
 <!-- <trkpt lat="70.6796266" lon="23.6723991"><time>2002-12-22T21:43:52Z</time><extensions><error>desc</error></extensions></trkpt> -->
 <!-- <trkpt lat="70.6796266" lon="23.6723991"><time>2002-12-22T21:43:54Z</time><extensions><error>desc</error></extensions></trkpt> -->
+<!-- 20021222T214351-20021222T214354: CO: No signal \x7D\x7D\x7D -->
 <trkpt lat="70.6800774" lon="23.6757566"><time>2002-12-22T21:44:45Z</time></trkpt>
 </trkseg>
 <trkseg>
Index: gpst
===================================================================
--- gpst	(revision 2020)
+++ gpst	(working copy)
@@ -273,6 +273,7 @@
                 push(@first_lines, $_);
             }
             s/^# error // && ($Dat{'error'} = "error");
+            s/^# (.+)// && ($Dat{'error'} = $1);
             s/^# ?// && ($Dat{'error'} = "desc");
             $xml_data = "";
             if (m#^<(e?tp)\b(.*?)>(.*?)</(e?tp)>\s*$#) {
