Index: create_funcs.sql
===================================================================
--- create_funcs.sql	(revision 2830)
+++ create_funcs.sql	(working copy)
@@ -142,9 +142,6 @@
     curr_id integer;
     currpoint point;
 BEGIN
-    -- FOR curr IN SELECT * FROM wayp_new LOOP
-    --     NULL;
-    -- END LOOP;
     UPDATE wayp_new SET coor = point(
         round(coor[0]::numeric, 6),
         round(coor[1]::numeric, 6)
@@ -154,22 +151,25 @@
         IF curr_id IS NOT NULL THEN
             RAISE NOTICE 'curr_id er ikke null: %', curr_id;
             currpoint = (SELECT coor FROM wayp_new WHERE id = curr_id);
-            IF (SELECT coor FROM wayp WHERE coor[0] = currpoint[0] AND coor[1] = currpoint[1]) IS NOT NULL THEN
+            IF (SELECT coor FROM wayp WHERE coor[0] = currpoint[0] AND coor[1] = currpoint[1] LIMIT 1) IS NOT NULL THEN
                 RAISE NOTICE '% finnes allerede i wayp', currpoint;
                 INSERT INTO wayp_rej SELECT * FROM wayp_new WHERE id = curr_id;
             ELSE
                 RAISE NOTICE '% er ikke i wayp', currpoint;
                 INSERT INTO wayp SELECT * FROM wayp_new WHERE id = curr_id;
-                PERFORM update_trackpoint(currpoint);
+                RAISE NOTICE 'Før UPDATE logg...';
+                UPDATE logg SET upd = TRUE WHERE upd IS NOT TRUE AND coor <-> currpoint < 0.05;
+                RAISE NOTICE '...UPDATE logg er ferdig, upd-count = %', (SELECT count(*) FROM logg WHERE upd = TRUE);
             END IF;
             DELETE FROM wayp_new WHERE id = curr_id;
-            -- COPY (SELECT name FROM wayp WHERE coor::varchar = currpoint::varchar)
-            --     TO STDOUT;
         ELSE
-            RAISE NOTICE 'currpoint er null';
+            RAISE NOTICE 'curr_id er null';
             EXIT;
         END IF;
     END LOOP;
+    RAISE NOTICE'UPDATE logg med clname og cldist...';
+    UPDATE logg SET sted = clname(coor), dist = cldist(coor), upd = FALSE WHERE upd = TRUE;
+    RAISE NOTICE '...ferdig.';
 END;
 $$ LANGUAGE plpgsql; -- }}}
 
Index: create_index.sql
===================================================================
--- create_index.sql	(revision 2830)
+++ create_index.sql	(working copy)
@@ -6,6 +6,7 @@
 CREATE INDEX log_ele_idx on logg(ele);
 CREATE INDEX log_sted_idx on logg(sted);
 CREATE INDEX log_dist_idx on logg(dist);
+CREATE INDEX log_upd_idx on logg(upd);
 
 -- DROP INDEX begin_idx;
 -- DROP INDEX end_idx;
Index: create_table.sql
===================================================================
--- create_table.sql	(revision 2830)
+++ create_table.sql	(working copy)
@@ -6,7 +6,9 @@
     ele numeric,
     sted text,
     dist numeric(8, 5),
-    description text
+    description text,
+    upd boolean DEFAULT TRUE,
+    id serial
 );
 
 CREATE TABLE wayp (
