$Id$

Index: src/edit.c
===================================================================
RCS file: /cvsroot/vim/vim7/src/edit.c,v
retrieving revision 1.141
diff -u -r1.141 edit.c
--- src/edit.c	6 Aug 2008 16:56:55 -0000	1.141
+++ src/edit.c	30 Aug 2008 08:23:06 -0000
@@ -6175,6 +6175,7 @@
 	textwidth = W_WIDTH(curwin) - 1;
 	if (textwidth > 79)
 	    textwidth = 79;
+	textwidth = 72; /* Added by sunny@sunbase.org -- always use 72 if not defined. */
     }
     return textwidth;
 }
Index: src/ex_cmds.c
===================================================================
RCS file: /cvsroot/vim/vim7/src/ex_cmds.c,v
retrieving revision 1.113
diff -u -r1.113 ex_cmds.c
--- src/ex_cmds.c	6 Aug 2008 13:03:07 -0000	1.113
+++ src/ex_cmds.c	30 Aug 2008 08:23:06 -0000
@@ -172,14 +172,14 @@
 	/*
 	 * if 'textwidth' set, use it
 	 * else if 'wrapmargin' set, use it
-	 * if invalid value, use 80
+	 * if invalid value, use 72
 	 */
 	if (width <= 0)
 	    width = curbuf->b_p_tw;
 	if (width == 0 && curbuf->b_p_wm > 0)
 	    width = W_WIDTH(curwin) - curbuf->b_p_wm;
 	if (width <= 0)
-	    width = 80;
+	    width = 72; /* Changed from 80 to 72 by sunny@sunbase.org */
     }
 
     if (u_save((linenr_T)(eap->line1 - 1), (linenr_T)(eap->line2 + 1)) == FAIL)
