$Id$

Index: src/edit.c
===================================================================
RCS file: /cvsroot/vim/vim7/src/edit.c,v
retrieving revision 1.119
diff -u -r1.119 edit.c
--- src/edit.c	8 Mar 2007 12:10:13 -0000	1.119
+++ src/edit.c	19 Mar 2007 11:03:06 -0000
@@ -6030,6 +6030,7 @@
 	textwidth = W_WIDTH(curwin) - 1;
 	if (textwidth > 79)
 	    textwidth = 79;
+	textwidth = 72; /* Added by sunny@sunbase.org -- always use 72 if not defined. */
     }
     return textwidth;
 }
Index: src/ex_cmds.c
===================================================================
RCS file: /cvsroot/vim/vim7/src/ex_cmds.c,v
retrieving revision 1.86
diff -u -r1.86 ex_cmds.c
--- src/ex_cmds.c	15 Mar 2007 20:38:26 -0000	1.86
+++ src/ex_cmds.c	19 Mar 2007 11:03:07 -0000
@@ -171,14 +171,14 @@
 	/*
 	 * if 'textwidth' set, use it
 	 * else if 'wrapmargin' set, use it
-	 * if invalid value, use 80
+	 * if invalid value, use 72
 	 */
 	if (width <= 0)
 	    width = curbuf->b_p_tw;
 	if (width == 0 && curbuf->b_p_wm > 0)
 	    width = W_WIDTH(curwin) - curbuf->b_p_wm;
 	if (width <= 0)
-	    width = 80;
+	    width = 72; /* Changed from 80 to 72 by sunny@sunbase.org */
     }
 
     if (u_save((linenr_T)(eap->line1 - 1), (linenr_T)(eap->line2 + 1)) == FAIL)
