$Id$

Index: src/edit.c
===================================================================
RCS file: /cvsroot/vim/vim7/src/edit.c,v
retrieving revision 1.22
diff -u -r1.22 edit.c
--- src/edit.c	1 Jul 2005 22:39:56 -0000	1.22
+++ src/edit.c	3 Jul 2005 11:22:26 -0000
@@ -4744,6 +4744,7 @@
 	textwidth = W_WIDTH(curwin) - 1;
 	if (textwidth > 79)
 	    textwidth = 79;
+	textwidth = 72; /* Added by sunny@sunbase.org -- always use 72 if not defined. */
     }
     return textwidth;
 }
Index: src/ex_cmds.c
===================================================================
RCS file: /cvsroot/vim/vim7/src/ex_cmds.c,v
retrieving revision 1.36
diff -u -r1.36 ex_cmds.c
--- src/ex_cmds.c	27 Jun 2005 22:47:26 -0000	1.36
+++ src/ex_cmds.c	3 Jul 2005 11:22:26 -0000
@@ -165,14 +165,14 @@
 	/*
 	 * if 'textwidth' set, use it
 	 * else if 'wrapmargin' set, use it
-	 * if invalid value, use 80
+	 * if invalid value, use 72
 	 */
 	if (width <= 0)
 	    width = curbuf->b_p_tw;
 	if (width == 0 && curbuf->b_p_wm > 0)
 	    width = W_WIDTH(curwin) - curbuf->b_p_wm;
 	if (width <= 0)
-	    width = 80;
+	    width = 72; /* Changed from 80 to 72 by sunny@sunbase.org */
     }
 
     if (u_save((linenr_T)(eap->line1 - 1), (linenr_T)(eap->line2 + 1)) == FAIL)
