$Id$

Index: src/feature.h
===================================================================
RCS file: /cvsroot/vim/vim7/src/feature.h,v
retrieving revision 1.32
diff -u -r1.32 feature.h
--- src/feature.h	1 Nov 2006 17:11:38 -0000	1.32
+++ src/feature.h	19 Mar 2007 11:00:36 -0000
@@ -962,7 +962,7 @@
  *			a modifed version of Vim.
  *			Also from the "--with-modified-by" configure argument.
  */
-/* #define MODIFIED_BY "John Doe" */
+#define MODIFIED_BY "Øyvind A. Holm <sunny@sunbase.org>"
 
 /*
  * Machine dependent:
